source/geometries/JaszczakPhantom.cc

```diff
@@ -0,0 +1,200 @@
+// ----------------------------------------------------------------------------
+// petalosim | JaszczakPhantom.cc
+//
+// This class implements the geometry of a Jaszczak phantom, filled with water.
+//
+// The PETALO Collaboration
+// ----------------------------------------------------------------------------
+
+#include "JaszczakPhantom.h"
+
+#include "nexus/FactoryBase.h"
+#include "nexus/Visibilities.h"
+#include "nexus/CylinderPointSampler2020.h"
+
+#include <G4Tubs.hh>
+#include <G4Orb.hh>
+#include <G4LogicalVolume.hh>
+#include <G4Material.hh>
+#include <G4NistManager.hh>
+#include <G4PVPlacement.hh>
+#include <G4RotationMatrix.hh>
+#include <G4VisAttributes.hh>
+#include <G4GenericMessenger.hh>
+#include <G4TransportationManager.hh>
+#include <G4Navigator.hh>
+
+using namespace nexus;
+
+REGISTER_CLASS(JaszczakPhantom, GeometryBase)
+
+JaszczakPhantom::JaszczakPhantom(): GeometryBase(),
+                                    bckg_activity_(1),
+                                    sphere_activity_(4),
+                                    rod_activity_(4),
+                                    cylinder_inner_diam_(216*mm),
+                                    cylinder_height_(186*mm),
+                                    cylinder_thickn_(3.2*mm),
+                                    sphere1_d_(9.5*mm),
+                                    sphere2_d_(12.7*mm),
+                                    sphere3_d_(15.9*mm),
+                                    sphere4_d_(19.1*mm),
+                                    sphere5_d_(25.4*mm),
+                                    sphere6_d_(31.8*mm),
+                                    sphere_height_(127*mm),
+                                    rod1_d_(3.2*mm),
+                                    rod2_d_(4.8*mm),
+                                    rod3_d_(6.4*mm),
+                                    rod4_d_(7.9*mm),
+                                    rod5_d_(9.5*mm),
+                                    rod6_d_(11.1*mm),
+                                    rod_height_(88*mm)
+{
+  // Messenger
+  msg_ = new G4GenericMessenger(this, "/Geometry/Jaszczak/",
+                                "Control commands of geometry Jaszczak.");
+  msg_->DeclareProperty("bckg_activity",   bckg_activity_,   "Activity of the background of the phantom");
+  msg_->DeclareProperty("sphere_activity", sphere_activity_, "Activity of the spheres");
+  msg_->DeclareProperty("rod_activity",    rod_activity_,    "Activity of the rods");
+
+  /// Initializing the geometry navigator (used in vertex generation)
+  geom_navigator_ =
+    G4TransportationManager::GetTransportationManager()->GetNavigatorForTracking();
+}
+
+
+JaszczakPhantom::~JaszczakPhantom()
+{
+}
+
+
+void JaszczakPhantom::Construct()
+{
+  auto cyl_name = "CYLINDER";
+  auto cylinder_solid =
+    new G4Tubs(cyl_name, 0, (cylinder_inner_diam_+2*cylinder_thickn_)/2,
+               (cylinder_height_+2*cylinder_thickn_)/2., 0, twopi);
```

Comment:
In `cylinder_height_` is it not already included `cylinder_thickn_`? I understand that in the case of `cylinder_inner_diam_` it must be added since we are referring to the inner diameter, but in height I was not sure.

Reply:
The arguments of `G4Tubs` are the semi-length and the radius of the cylinder. Since we are creating a cylinder made of aluminum and then placing water inside it, we need to make it slightly bigger than the water volume in every dimension, right? In this case `cylinder_height_` is the length of the water volume inside. If you think it makes it clearer, I can use `cylinder_internal_height`.